src/transformers/models/vit/convert_vit_timm_to_pytorch.py

```diff
@@ -0,0 +1,230 @@
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc1.bias", "vit.encoder.layer." + str(i) + ".intermediate.dense.bias")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc2.weight", "vit.encoder.layer." + str(i) + ".output.dense.weight")
+        )
+        rename_keys.append(
+            ("blocks." + str(i) + ".mlp.fc2.bias", "vit.encoder.layer." + str(i) + ".output.dense.bias")
+        )
+
+    # projection layer + position embeddings
+    rename_keys.extend(
+        [
+            ("cls_token", "vit.embeddings.cls_token"),
+            ("patch_embed.proj.weight", "vit.embeddings.patch_embeddings.projection.weight"),
+            ("patch_embed.proj.bias", "vit.embeddings.patch_embeddings.projection.bias"),
+            ("pos_embed", "vit.embeddings.position_embeddings"),
+        ]
+    )
+
+    # classification head
+    rename_keys.extend(
+        [
+            ("head.weight", "classifier.weight"),
+            ("head.bias", "classifier.bias"),
+            ("norm.weight", "layernorm.weight"),
+            ("norm.bias", "layernorm.bias"),
+        ]
+    )
+
+    # to do: add base model support
+    # if just the base model, we should remove "vit" from all keys
+    if base_model:
+        pass
+
+    return rename_keys
+
+
+# we split up the matrix of each encoder layer into queries, keys and values
+def read_in_q_k_v(state_dict, config, base_model=False):
+    for i in range(config.num_hidden_layers):
+        # read in weights + bias of input projection layer (in timm, this is a single matrix + bias)
+        in_proj_weight = state_dict.pop("blocks." + str(i) + ".attn.qkv.weight")
+        in_proj_bias = state_dict.pop("blocks." + str(i) + ".attn.qkv.bias")
+        # next, add query, keys and values (in that order) to the state dict
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.query.weight"] = in_proj_weight[
+            : config.hidden_size, :
+        ]
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.query.bias"] = in_proj_bias[: config.hidden_size]
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.key.weight"] = in_proj_weight[
+            config.hidden_size : config.hidden_size * 2, :
+        ]
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.key.bias"] = in_proj_bias[
+            config.hidden_size : config.hidden_size * 2
+        ]
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.value.weight"] = in_proj_weight[
+            -config.hidden_size :, :
+        ]
+        state_dict["vit.encoder.layer." + str(i) + ".attention.self.value.bias"] = in_proj_bias[-config.hidden_size :]
+
+    # to do: add base model support
+    if base_model:
+        pass
+
+
+def remove_classification_head_(state_dict):
+    ignore_keys = [
+        "norm.weight",
+        "norm.bias",
+        "head.weight",
+        "head.bias",
```

Comment:
This can all fit in one line.

Reply:
I know, it's `make style` that does this.

Reply:
Just tested locally and it did not change the line
```
    ignore_keys = ["norm.weight", "norm.bias", "head.weight", "head.bias"]
```